nibabel/freesurfer/mghformat.py

```diff
@@ -281,7 +281,7 @@ def set_data_dtype(self, datatype):
         try:
             code = self._data_type_codes[datatype]
         except KeyError:
-            raise MGHError('datatype dtype "%s" not recognized' % datatype)
+            raise HeaderDataError('datatype dtype "%s" not recognized' % datatype)
```

Comment:
With the added supported_np_types check in spatialimages the missing HeaderDataError produced errors in the test_image_api and test_image_load_save tests. HeaderDataError seems to be the "correct" error to raise here.